util-taglib/src/com/liferay/taglib/util/InlineUtil.java

```diff
@@ -33,18 +35,40 @@ public static String buildDynamicAttributes(
 
 		StringBundler sb = new StringBundler(dynamicAttributes.size() * 4);
 
+		List<String> attributesList = Arrays.asList(_HTML_BOOLEAN_ATTRIBUTES);
+
 		for (Map.Entry<String, Object> entry : dynamicAttributes.entrySet()) {
 			String key = entry.getKey();
 
 			if (!key.equals("class")) {
-				sb.append(key);
-				sb.append("=\"");
-				sb.append(String.valueOf(entry.getValue()));
-				sb.append("\" ");
+				if (attributesList.contains(key)) {
+					boolean value = (boolean)entry.getValue();
```

Comment:
@izaera, do you know if we have a specific SF-sanctioned way to do this type of casting?

Reply:
I doubt that's going to work, in fact, because the value can probably be a `String`:

```
 <tag value="false"/>
``` 

is `String`

```
 <tag value="<%= false %>"/>
``` 
is a `Boolean`.

And I suppose we want to support the first case too.

So, the best way would probably be `Boolean.valueOf(entry.getValue())`.

Reply:
Thanks @izaera , yes with the code as it is right now doing this 

```jsp
<clay:button disabled="false"/>
```

Will throw an exception, so I'll update that and test.

Reply:
Updated [here](https://github.com/liferay-frontend/liferay-portal/commit/2a47bdd5188926740714998cfb7a6ab1e1bbbb89)